bot.js

```diff
@@ -59,109 +82,181 @@ const parseCommand = (text) => {
+                break;
+            default:
+                const text = Object.keys(diff).toSortedSkills().map((skill) => {
                     const levelsGained = diff[skill];
-                    sendUpdateMessage(trackingChannel,
-                        `**${player}** has gained `
-                            + (levelsGained === 1 ? 'a level' : `**${levelsGained}** levels`)
-                            + ` in **${skill}** and is now level **${newLevels[skill]}**`,
-                        skill);
-                    break;
-                default:
-                    const text = Object.keys(diff).toSortedSkills().map((skill) => {
-                        const levelsGained = diff[skill];
-                        return `${levelsGained === 1 ? 'a level' : `**${levelsGained}** levels`} in **${skill}** and is now level **${newLevels[skill]}**`;
-                    }).join('\n');
-                    sendUpdateMessage(trackingChannel, `**${player}** has gained...\n${text}`, 'overall');
-                    break;
+                    return `${levelsGained === 1 ? 'a level' : `**${levelsGained}** levels`} in **${skill}** and is now level **${newLevels[skill]}**`;
+                }).join('\n');
+                sendUpdateMessage(trackingChannel, `**${player}** has gained...\n${text}`, 'overall');
+                break;
+        }
+    }
+    // If not spoofing the diff, update player's levels
+    if (!spoofedDiff) {
+        levels[player] = newLevels;
+        lastUpdate[player] = new Date();
+    }
+};
+
+const updateKillCounts = (player, killCounts, spoofedDiff) => {
+    // If channel is set and user already has bosses tracked
+    if (trackingChannel && bosses.hasOwnProperty(player)) {
+        // Compute diff for each boss
+        let diff;
+        try {
+            if (spoofedDiff) {
+                diff = spoofedDiff;
+                Object.keys(diff).forEach((boss) => {
+                    killCounts[boss] += diff[boss];
+                });
+            } else {
+                diff = computeDiff(bosses[player], killCounts);
             }
+        } catch (err) {
+            log.push(`Failed to compute boss KC diff for player ${player}: ${err.toString()}`);
+            return;
         }
-        // If not spoofing the diff, update player's levels
-        if (!spoofedDiff) {
-            levels[player] = newLevels;
-            lastUpdate[player] = new Date();
+        if (!diff) {
+            return;
+        }
+        // Send a message showing all the incremented boss KCs
+        switch (Object.keys(diff).length) {
+            case 0:
+                break;
+            case 1:
+                const boss = Object.keys(diff)[0];
+                const killCountIncrease = diff[boss];
+                sendUpdateMessage(trackingChannel,
+                    `**${player}** has killed **${boss}** ${killCountIncrease} `
+                        + (killCountIncrease === 1 ? 'more time' : `more times`)
```

Comment:
Perhaps phrase the single kill text to something like `Vijuhas has killed KBD again and is now at 100000 kills`. For some reason, saying `Vijuhas has killed KBD 1 more time...` doesn't sound too smooth to me. Or for super epic random points you can have a random phrase like `has killed KBD again` or `has killed KBD once more`. Or if it's the first kill, you can specialize it to say `has killed KBD for the first time`. A lot of creative ways to do this.